cores/esp32/esp32-hal-ledc.c

```diff
@@ -58,15 +60,25 @@ static bool ledcDetachBus(void * bus){
 
 bool ledcAttachChannel(uint8_t pin, uint32_t freq, uint8_t resolution, uint8_t channel)
 {
-    if (channel >= LEDC_CHANNELS || resolution > LEDC_MAX_BIT_WIDTH)
-    {
-        log_e("Channel %u is not available! (maximum %u) or bit width too big (maximum %u)", channel, LEDC_CHANNELS, LEDC_MAX_BIT_WIDTH);
+    if(channel >= LEDC_CHANNELS || ledc_handle.used_channels & (1UL << channel)){
+        log_e("Channel %u is not available (maximum %u) or already used!", channel, LEDC_CHANNELS);
+        return false;
+    }
+    
```

Comment:
```suggestion
    if (freq == 0) {
        log_e("LEDC pin %u - frequency can't be zero.", pin);
    }
```
frequency can't be zero as well.  IDF will fail with BAD PARAMETER error.

Reply:
This verification (freq != 0 && resoution != 0) shall also be done in
`uint32_t ledcChangeFrequency(uint8_t pin, uint32_t freq, uint8_t resolution)`

Reply:
in function `ledcChangeFrequency()`, it may better to change/add this:

``` cpp
        if (resolution == 0 || resolution > LEDC_MAX_BIT_WIDTH){  
            log_e("LEDC pin %u - resolution is zero or it is too big (maximum %u)", pin, LEDC_MAX_BIT_WIDTH);
            return 0;
        }
        if (freq == 0) {
            log_e("LEDC pin %u - frequency can't be zero.", pin);
            return 0;
        }
```